video/graphics.h

```diff
@@ -318,6 +356,24 @@ void plotLine(bool omitFirstPoint = false, bool omitLastPoint = false) {
 	}
 }
 
+// Fill horizontal line
+//
+void fillHorizontalLine(bool scanLeft, bool match, RGB888 matchColor) {
+	canvas->waitCompletion(false);
```

Comment:
with this implementation we need to force the queue to flush, otherwise our scan routines will not see lines that have recently been drawn.  this is inefficient, especially if sprites are on-screen.

moving this functionality into vdp-gl would allow for the code to be written so line can be drawn as part of the queue processing, thus removing the need to flush the queue.

indeed if this is _not_ done as part of the queue processing then lines drawn this way might interact with sprites in unintended ways